aeron-cluster/src/main/java/io/aeron/cluster/ConsensusModuleAgent.java

```diff
@@ -3598,4 +3611,31 @@ public String toString()
             "election=" + election +
             '}';
     }
+
+    private static final class UnknownConsensusModuleExtension implements ConsensusModuleExtension
+    {
+
+        private static final UnknownConsensusModuleExtension INSTANCE = new UnknownConsensusModuleExtension();
+        private UnknownConsensusModuleExtension()
+        {
+        }
+
+        @Override
+        public int supportedSchemaId()
+        {
+            return 0;
+        }
+
+        @Override
+        public ControlledFragmentHandler.Action onFragment(
+            final int schemaId,
+            final int templateId,
+            final DirectBuffer buffer,
+            final int offset,
+            final int length,
+            final Header header)
+        {
+            throw new ClusterException("expected schemaId=" + MessageHeaderDecoder.SCHEMA_ID + ", actual=" + schemaId);
```

Comment:
Perform null check in Agent rather than have another implementation that could defeat monomorphic optimisation.

Reply:
removed unknown schema impl